### ISSUE TITLE
dataSize sometimes contains negative values

### ISSUE THREAD

**Issue by morningkyle** [swharden/ScottPlot#144]
Tested on master branch.
[void Settings::Resize(int width, int height)](https://github.com/swharden/ScottPlot/blob/master/src/ScottPlot/Settings.cs) initializes dataSize to negative values during "quickstart wpf" demo executable startup.  The unexpected negative width and height of dataSize can be caught by adding a few assertions inside Setting::Resize() as below.  Although no real exception triggered at run-time, it might indicate some cleanups needed in future.

```
public void Resize(int width, int height)
{
    // TODO: data origin should be calculated at render time, not now.
    figureSize = new Size(width, height);
    dataOrigin = new Point(layout.paddingBySide[0], layout.paddingBySide[3]);
    int dataWidth = figureSize.Width - layout.paddingBySide[0] - layout.paddingBySide[1];
    int dataHeight = figureSize.Height - layout.paddingBySide[2] - layout.paddingBySide[3];
    Debug.Assert(dataWidth >= 0);
    Debug.Assert(dataHeight >= 0);
    dataSize = new Size(dataWidth, dataHeight);
 }
```

**Comment by morningkyle** [comment]
The above commit is my attempt to fix dataSize initialization.  It should be safe to clamp dataSize.Width/Height  above zero.

**Comment by swharden** [comment]
Hi @morningkyle, thanks for looking into this one!

I'm hesitant to change how `dataSize` is calculated. I think the intent of this variable is to hold the "size of the data area given the current figure size and various layout options". Every time the figure is resized, new bitmaps are created in memory, and `dataSize` indicates how large the data area bitmap should be. 

If the figure is very small (too small to show data), `dataSize` will contain negative values, and this is how the system knows not to allocate memory for the data area bitmap. For example, this check exists in the `InitializeBitmaps()` function:

https://github.com/swharden/ScottPlot/blob/8802e583148a3918f7ee2dd8c940b82f411cd67f/src/ScottPlot/Plot.cs#L77-L81

For this reason, it may not be safe to clamp these values above zero.

I wouldn't invest too much effort optimizing the layout system right now (such as the `dataSize` variable), as I plan to rewrite the layout system soon (#134) very soon. It is my hope that this whole layout procedure (including bitmap size calculations) will be greatly simplified after I do.

**Comment by morningkyle** [comment]
I edited my last comment to "_It should be safe to clamp dataSize.Width/Height >= zero._" 

Conceptually, it's hard to understand what negative width and height mean to an image/bitmap.  If we want to know when NOT to allocate memory for bitmap, 0 width/height works intuitively(and makes good sense). And your above code should still work:)

I happened to look into this part because I tried to understand how the canvas resizing works.  It's good to know you are working on improving&simplifying this part.  Thanks!
